SharpCompress/Writer/Zip/ZipWriter.cs

```diff
@@ -85,15 +51,16 @@ public override void Write(string entryPath, Stream source, DateTime? modificati
             Write(entryPath, source, modificationTime, null);
         }
 
-        public void Write(string entryPath, Stream source, DateTime? modificationTime, string comment)
+        public void Write(string entryPath, Stream source, DateTime? modificationTime, string comment, CompressionInfo compressionInfo = null)
         {
-            using (Stream output = WriteToStream(entryPath, modificationTime, comment))
+            var explicitCompressionInfo = compressionInfo != null ? new ZipCompressionInfo(compressionInfo) : this.zipCompressionInfo;
+            using (Stream output = WriteToStream(entryPath, modificationTime, comment, explicitCompressionInfo))
             {
                 source.TransferTo(output);
             }
         }
 
-        public Stream WriteToStream(string entryPath, DateTime? modificationTime, string comment)
+        public Stream WriteToStream(string entryPath, DateTime? modificationTime, string comment, ZipCompressionInfo explicitCompressionInfo = null)

```

Comment:
I think it would be better to pass `CompressionInfo` as an optional parameter here instead. The `ZipCompressionInfo` is specific to the ZIP format and should be kept for internal usage only.